### PR TITLE
Release

### PR DESCRIPTION
**Pull request by marthacryan**
# Adds Tests that will run only on pull requests to main
Adds tests that go through a more involved workflow on the test pypi release of mitosheet when a PR is made to the main branch.
* adds a test that follows the instructions here: https://github.com/mito-ds/every-feature-everywhere-all-at-once/tree/main/Demos/vanguard-fund-performance#launch-the-streamlit-app

**Note: these changes don't incorporate the newest updates to testing from dev because for testing reasons this needed to be a PR to main.**

### CODE REVIEW

File: tests/streamlit_ui_tests/demos/vanguard.spec.ts [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+import { awaitResponse, clickTab, getMitoFrame } from '../utils';
+
+const createNewColumnWithName = async (page: Page, mito: FrameLocator, name: string) => {
+    await clickTab(page, mito, 'Home');
+    await mito.locator('.mito-toolbar-button', { hasText: 'Insert' }).click();
+    await mito.locator('.endo-column-header-container', { hasText: /new-column/ }).dblclick();
+    await mito.locator('.endo-column-header-container input').fill(name);
+    await mito.locator('.endo-column-header-container input').press('Enter', { delay: 100 });
+    await expect(mito.locator('.endo-column-header-container', { hasText: name })).toBeVisible();
+};
```
Comment (naterush) [inline tests/streamlit_ui_tests/demos/vanguard.spec.ts:11]: Don't we already have this utility?

File: .github/workflows/prerelease-tests.yml [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+            # Install necessary node packages
+            npm install
+            npx playwright install chromium webkit firefox chrome
+
+        - name: Start a server and run tests
+          run: |
+            cd tests
+            source venv/bin/activate
+            jupyter lab --config jupyter_server_test_config.py &
+            npm run test:jupyterlab
```
Comment (naterush) [inline .github/workflows/prerelease-tests.yml:52]: This is just running jupyter lab tests, no?

File: tests/streamlit_ui_tests/demos/vanguard.spec.ts [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+    await createNewColumnWithName(page, mito, 'Portfolio Manager');
+    await mito.locator('.mito-grid-cell[mito-col-index="3"]').first().dblclick();
+    await mito.locator('#cell-editor-input').fill('=VLOOKUP(');
+    await mito.locator('.mito-grid-cell[mito-col-index="1"]').first().click();
+    await mito.locator('#cell-editor-input').pressSequentially(', ');
+    await mito.locator('.tab', { hasText: 'fund_info' }).click();
+    await mito.getByText('Fund Name').click();
+    await mito.getByText('Ongoing Charge').click({ modifiers: ['Shift'] });
+    await mito.locator('#cell-editor-input').pressSequentially(', 2)');
+    await mito.locator('#cell-editor-input').press('Enter');
```
Comment (naterush) [inline tests/streamlit_ui_tests/demos/vanguard.spec.ts:38]: Not now, but .... this should be a utility :)

File: .github/workflows/prerelease-tests.yml [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+          npm run test:streamlit:demo -- --project="${{ matrix.project }}"
+      - name: Start a server and run tests (windows)
+        if: matrix.os == 'windows-latest'
+        run: |
+          cd tests
+          .\venv\Scripts\Activate.ps1
+
+          git clone https://github.com/mito-ds/every-feature-everywhere-all-at-once.git
+          cd every-feature-everywhere-all-at-once
+          Start-Job { streamlit run automation-app.py --server.port 8555 } -WorkingDirectory (Get-Location)
```
Comment (naterush) [inline .github/workflows/prerelease-tests.yml:154]: Do we need to run a test on windows that the server is running?

**Review by naterush — COMMENTED** [review]
Overall, very reasonable. LGTM when tests pass

File: .github/workflows/prerelease-tests.yml (hunk 2 above)
Comment (marthacryan) [reply .github/workflows/prerelease-tests.yml:52]: Yes, but on the test pypi version.

File: tests/streamlit_ui_tests/demos/vanguard.spec.ts (hunk 1 above)
Comment (marthacryan) [reply tests/streamlit_ui_tests/demos/vanguard.spec.ts:11]: Yes but this is based off of main, so it doesn't have the useful utilities yet!

File: .github/workflows/prerelease-tests.yml (hunk 4 above)
Comment (marthacryan) [reply .github/workflows/prerelease-tests.yml:154]: I think I just grabbed this from whatever we're currently doing. I guess the tests will catch if the server doesn't start up? Not totally suer why we need that in general

File: .github/workflows/prerelease-tests.yml (hunk 2 above)
Comment (marthacryan) [reply .github/workflows/prerelease-tests.yml:52]: Also: I wanted to add a more in-depth jupyterlab demo test but ran out of time!

**Pull request merged**